mls-rs/src/group/proposal_cache.rs

```diff
@@ -299,17 +295,12 @@ fn has_ref(proposals: &ProposalBundle, reference: &ProposalRef) -> bool {
 fn rejected_proposals(
     all_proposals: ProposalBundle,
     accepted_proposals: &ProposalBundle,
-    sender: &Sender,
-) -> Vec<(ProposalRef, Proposal)> {
+) -> Vec<ProposalInfo<Proposal>> {
     all_proposals
         .into_proposals()
-        .filter_map(|p| match p.source {
-            ProposalSource::ByReference(r)
-                if !has_ref(accepted_proposals, &r) && &p.sender == sender =>
-            {
-                Some((r, p.proposal))
-            }
-            _ => None,
+        .filter(|p| {
+            matches!(p.source, ProposalSource::ByReference(ref r) if !has_ref(accepted_proposals, r)
```

Comment:
It seems that rejected proposals will now contain proposals from all members instead of only the one sending or receiving the commit? Is that an intentional change of behavior?

Reply:
Yes, and that's what I like most about this PR ;-)

Reply:
Yeah, that is required in order to use this within the `ExternalClient` and it makes sense because you can always filter it post receiving this information.